src/main/java/seedu/address/logic/parser/FindExpressionParser.java

```diff
@@ -0,0 +1,288 @@
+     * using the OR operator. For example, in a query "n/Bob OR (t/friends AND t/colleagues)",
+     * both "n/Bob" and "(t/friends AND t/colleagues)" are terms combined to form an expression.
+     * </p>
+     *
+     * @return The parsed expression node.
+     */
+    private ExprNode expression() throws ParseException {
+        ExprNode node = term();
+
+        while (match(Token.Type.OR)) {
+            Token op = consume(Token.Type.OR);
+            ExprNode right = term();
+            node = new BinaryOpNode(node, FindExpressionOperator.OR, right);
+        }
+
+        return node;
+    }
+
+    /**
+     * Parses and constructs a "term" node from the current list of tokens.
+     * <p>
+     * In the context of parsers and grammars, a "term" is a mid-level construct that consists of one or more
+     * "factors". The factors within a term are usually combined using the AND operator. For instance,
+     * in a query "n/Bob AND t/friends", both "n/Bob" and "t/friends" are factors combined to create a term.
+     * </p>
+     *
+     * @return The parsed term node.
+     */
+    private ExprNode term() throws ParseException {
+        ExprNode node = factor();
+
+        while (match(Token.Type.AND)) {
+            Token op = consume(Token.Type.AND);
+            ExprNode right = factor();
+            node = new BinaryOpNode(node, FindExpressionOperator.AND, right);
+        }
+
+        return node;
+    }
+
+    /**
+     * Parses and constructs a "factor" node from the current list of tokens.
+     * <p>
+     * A "factor" is a foundational construct in this parser. It's the most granular unit and represents
+     * a single condition or criteria in the query. Factors are building blocks for terms and expressions.
+     * </p>
+     *
+     * @return The parsed factor node.
+     */
+    private ExprNode factor() throws ParseException {
+        if (match(Token.Type.NOT)) {
+            consume(Token.Type.NOT);
+            ExprNode node = factor();
+            return new NotNode(node);
+        } else if (match(Token.Type.LPAREN)) {
+            consume(Token.Type.LPAREN);
+            ExprNode node = expression();
+            consume(Token.Type.RPAREN);
+            return node;
+        } else {
+            Token token = consume(Token.Type.CONDITION);
+            // check if text contains a slash and is a valid condition
+            if (!token.text.contains("/") || token.text.startsWith("/") || token.text.endsWith("/")) {
+                throw new ParseException("Invalid condition: " + token.text);
+            }
+            // split by slash but include slash in substrings
+            String[] parts = token.text.split("(?<=/)");
+            FindSupportedField field = FindSupportedField.fromPrefix(parts[0].trim().toLowerCase());
+            return new ConditionNode(field, parts[1].trim());
+        }
+    }
+
+    private boolean match(Token.Type type) {
```

Comment:
Consider renaming this to [sound more like a boolean](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=Boolean%20variables/methods%20should%20be%20named%20to%20sound%20like%20booleans)

Reply:
perhaps something like `nextTokenMatches`

Reply:
derp, I realise I don't need this at all since it's literally just a wrapper around check (it wasn't at first but I changed it while debugging, oops)